hubblestack/files/hubblestack_nova/misc.py

```diff
@@ -509,6 +509,256 @@ def check_core_dumps(reason=''):
+            error += ["User " + user_uid_dir[0] + " has invalid uid " + user_uid_dir[1]]
+    if error == []:
+        return True
+
+    return str(error)
+
+
+def check_users_home_directory_permissions(reason=''):
+    '''
+    Ensure users' home directories permissions are 750 or more restrictive
+    '''
+
+    users_dirs = _execute_shell_command("cat /etc/passwd | egrep -v '(root|halt|sync|shutdown)' | awk -F: '($7 != \"/sbin/nologin\") {print $1\" \"$6}'").strip()
+    users_dirs = users_dirs.split('\n') if users_dirs != "" else []
+    error = []
+    for user_dir in users_dirs:
+        user_dir = user_dir.split(" ")
+        if len(user_dir) < 2:
+                user_dir = user_dir + ['']*(2-len(user_dir))
+        if user_dir[1] is None or user_dir[1] == "":
+            error += ["User " + user_dir[0] + " does not have any home directoyr"]
+        else:
+            result = restrict_permissions(user_dir[1], "750")
+            if result is not True:
+                error += ["permission on home directory " + user_dir[1]  + " of user " + user_dir[0] + " is wrong: " + result]
+
+    if error == []:
+        return True
+
+    return str(error)
+
+
+def check_users_own_their_home(reason=''):
+    '''
+    Ensure users own their home directories
+    '''
+
+    users_uids_dirs = _execute_shell_command("cat /etc/passwd | awk -F: '{ print $1 \" \" $3 \" \" $6 }'").strip()
+    users_uids_dirs = users_uids_dirs.split('\n') if users_uids_dirs != "" else []
+    error = []
+    for user_data in users_uids_dirs:
+        user_uid_dir = user_data.strip().split(" ")
+        if len(user_uid_dir) < 3:
+                user_uid_dir = user_uid_dir + ['']*(3-len(user_uid_dir))
+        if user_uid_dir[1].isdigit():
+            if int(user_uid_dir[1]) >= 1000 and os.path.isdir(user_uid_dir[2]) and user_uid_dir[0] is not "nfsnobody":
+                owner = _execute_shell_command("stat -L -c \"%U\" \"" + user_uid_dir[2] + "\"")
+                if owner is not user_uid_dir[0]:
+                    error += ["The home directory " + user_uid_dir[2] + " of user " + user_uid_dir[0] + " is owned by " + owner]
+        else:
+            error += ["User " + user_uid_dir[0] + " has invalid uid " + user_uid_dir[1]]
+
+    if error == []:
+        return True
+
+    return str(error)
+
+
+def check_users_dot_files(reason=''):
+    '''
+    Ensure users' dot files are not group or world writable
+    '''
+
+    users_dirs = _execute_shell_command("cat /etc/passwd | egrep -v '(root|halt|sync|shutdown)' | awk -F: '($7 != \"/sbin/nologin\") {print $1\" \"$6}'").strip()
+    users_dirs = users_dirs.split('\n') if users_dirs != "" else []
+    error = []
+    for user_dir in users_dirs:
+        user_dir = user_dir.split(" ")
+        if len(user_dir) < 2:
+                user_dir = user_dir + ['']*(2-len(user_dir))
+        if user_dir[1] is None or user_dir[1] == "":
+            error += ["User " + user_dir[0] + " does not have any home directory"]
```

Comment:
Change error message as `["User " + user_dir[0] + " does not have valid home directory"]`